src/dftbp/timedep/timeprop.F90

```diff
@@ -1530,6 +1571,18 @@ subroutine updateH(this, H1, ints, H0, speciesAll, qq, q0, coord, orb, potential
       call qm2ud(qq)
     end if
 
+  #:if WITH_SCALAPACK
+    do iKS = 1, this%parallelKS%nLocalKS
+      iK = this%parallelKS%localKS(1, iKS)
+      iSpin = this%parallelKS%localKS(2, iKS)
+      if (this%tRealHS) then
+        call unpackHSRealBlacs(env%blacs, ints%hamiltonian(:,iSpin), neighbourList%iNeighbour,&
+            & nNeighbourSK, iSparseStart, img2CentCell, this%denseDesc, T2)
+        H1(:,:,iSpin) = cmplx(T2, 0.0_dp, dp)
+        ! TODO: add here the unpacking of H1 for kpoints
+      end if
+    end do
+  #:else
     do iKS = 1, this%parallelKS%nLocalKS
       iK = this%parallelKS%localKS(1, iKS)
       iSpin = this%parallelKS%localKS(2, iKS)
```

Comment:
We should leave away those lines as not needed.

Reply:
`iSpin` is always need and `iK` is also need in the complex case. Or there is a different way of accessing these indices?